### PR TITLE
enh(assert): Add new asserts

### PR DESCRIPTION
**Pull request by callapa**
## Description

Add new asserts (notEmpty and notNull).

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [ ] 20.04.x
- [ ] 20.10.x
- [x] 21.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Centreon/Domain/Common/Assertion/Assertion.php [hunk 1]
```diff
@@ -163,6 +163,42 @@ function (array $parameters) {
...
+    /**
+     * Assert that value is not null.
+     *
+     * @param mixed $value Value to test
+     * @param string|null $propertyPath Property's path (ex: Host::name)
+     * @throws \Assert\AssertionFailedException
+     */
+    public static function notNull($value, string $propertyPath = null): void
+    {
+        Assert::notEmpty(
```
Comment (jeremyjaouen) [inline src/Centreon/Domain/Common/Assertion/Assertion.php:193]: This shouldn't be `Assert::notNull` ?

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -15866,3 +15866,24 @@ msgstr "Erreur lors de la recherche des services"
 
 msgid "Error when searching for hosts"
 msgstr "Erreur lors de la recherche des hôtes"
+
+msgid "[%s] The value \"%s\" is too long, it should have no more than %d characters, but has %d characters"
+msgstr "[%s] La valeur est trop longue, elle ne devrait pas comporter plus de %d caractères mais n'en comporte que %d"
```
Comment (sc979) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15871]: [%s] La valeur est trop longue -> [%s] La valeur \"%s\" est trop longue

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -15866,3 +15866,24 @@ msgstr "Erreur lors de la recherche des services"
...
+
+msgid "[%s] The value \"%s\" is too short, it should have at least %d characters, but only has %d characters"
+msgstr "[%s] La valeur est trop courte, elle devrait comporter au moins %d caractères mais n'en a que %d"
```
Comment (sc979) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15874]: same

File: src/Centreon/Domain/Common/Assertion/Assertion.php (hunk 1 above)
Comment (sc979) [inline src/Centreon/Domain/Common/Assertion/Assertion.php:193]: Assert::notEmpty ?

File: src/Centreon/Domain/Common/Assertion/AssertionException.php [hunk 4]
```diff
@@ -138,4 +138,36 @@ public static function greaterOrEqualThan(int $value, int $limit, string $proper
...
         );
     }
+
+    /**
+     * Exception when the value is empty.
+     *
+     * @param string|null $propertyPath Property's path (ex: Host::name)
+     * @return self
+     */
+    public static function notEmpty(string $propertyPath = null): self
```
Comment (sc979) [inline src/Centreon/Domain/Common/Assertion/AssertionException.php:148]: if $propertyPath === null, what will be the [%s] value in the thrown message ?

**Review by jeremyjaouen — APPROVED** [review]

**Review by sc979 — DISMISSED** [review]

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -15855,14 +15855,29 @@ msgstr "EN ATTENTE"
...
-msgstr "[%s] La valeur \"%s\" est trop courte, elle devrait avoir au moins %d caractères, mais n'a que %d caractères"
-
 msgid "Error when searching for services"
 msgstr "Erreur lors de la recherche des services"
 
 msgid "Error when searching for hosts"
 msgstr "Erreur lors de la recherche des hôtes"
+
+msgid "[%s] The value \"%s\" is too long, it should have no more than %d characters, but has %d characters"
+msgstr "[%s] La valeur \"%s\" est trop longue, elle ne devrait pas comporter plus de %d caractères mais n'en comporte que %d"
```
Comment (adr-mo) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15865]:
```suggestion
msgstr "[%s] La valeur \"%s\" est trop longue, elle ne devrait pas comporter plus de %d caractères elle en comporte %d"
```

**Review by adr-mo — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**